x-pack/plugins/uptime/public/components/fleet_package/browser/simple_fields.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiFormRow, EuiFieldText, EuiFieldNumber } from '@elastic/eui';
+import { ConfigKeys, Validation } from '../types';
+import { useBrowserSimpleFieldsContext } from '../contexts';
+import { ComboBox } from '../combo_box';
+import { OptionalLabel } from '../optional_label';
+import { ScheduleField } from '../schedule_field';
+import { SourceField } from './source_field';
+
+interface Props {
+  validate: Validation;
+}
+
+export const BrowserSimpleFields = memo<Props>(({ validate }) => {
+  const { fields, setFields, defaultValues } = useBrowserSimpleFieldsContext();
+  const handleInputChange = ({ value, configKey }: { value: unknown; configKey: ConfigKeys }) => {
+    setFields((prevFields) => ({ ...prevFields, [configKey]: value }));
+  };
+
+  return (
+    <>
+      <EuiFormRow
+        id="syntheticsFleetScheduleField--number syntheticsFleetScheduleField--unit"
+        label={
+          <FormattedMessage
+            id="xpack.uptime.createPackagePolicy.stepConfigure.monitorIntegrationSettingsSection.monitorInterval"
+            defaultMessage="Monitor interval"
+          />
+        }
+        isInvalid={!!validate[ConfigKeys.SCHEDULE]?.(fields)}
+        error={
+          <FormattedMessage
+            id="xpack.uptime.createPackagePolicy.stepConfigure.monitorIntegrationSettingsSection.monitorInterval.error"
+            defaultMessage="Monitor interval is required"
+          />
+        }
+      >
+        <ScheduleField
+          onChange={(schedule) =>
+            handleInputChange({
+              value: schedule,
+              configKey: ConfigKeys.SCHEDULE,
+            })
+          }
+          number={fields[ConfigKeys.SCHEDULE].number}
+          unit={fields[ConfigKeys.SCHEDULE].unit}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.uptime.createPackagePolicy.stepConfigure.monitorIntegrationSettingsSection.browser.sourceType.label"
+            defaultMessage="Source Type"
+          />
+        }
+      >
+        <SourceField
+          onChange={useCallback(
```

Comment:
hmm inline memo, or call back hooks, i was tempted to do this other day , not sure if it has any implications, probably none.